src/main/java/com/flightstats/hub/webhook/LocalWebhookManager.java

```diff
@@ -22,38 +21,44 @@
 import java.util.function.Consumer;
 
 @Singleton
+@Slf4j
 public class LocalWebhookManager {
 
-    private static final Logger logger = LoggerFactory.getLogger(LocalWebhookManager.class);
+    private Map<String, WebhookLeader> localLeaders = new ConcurrentHashMap<>();
+
     private final KeyLockManager lockManager;
     @Inject
     @Named("Webhook")
     private Dao<Webhook> webhookDao;
     @Inject
     private Provider<WebhookLeader> v2Provider;
-    private Map<String, WebhookLeader> localLeaders = new ConcurrentHashMap<>();
+
+    private static int shutdownThreadCount;
 
     @Inject
-    public LocalWebhookManager() {
+    public LocalWebhookManager(@Named("Webhook") Dao<Webhook> webhookDao,
+                                   Provider<WebhookLeader> v2Provider,
+                               WebhookProperty webhookProperty) {
+        shutdownThreadCount = webhookProperty.getShutdownThreadCount();
```

Comment:
...does this mean that `runAndWait` can be called with a shutdown thread count of 0 if an instance is never instantiated? I'm not sure what the implications of that are, but it seems like it could cause problems.